tortoise/tortoise.go

```diff
@@ -904,40 +831,112 @@ func (t *turtle) layerCutoff() types.LayerID {
 	return t.last.Sub(t.Hdist)
 }
 
-func isUndecided(config Config, decided map[types.LayerID]types.BlockID, lid, last types.LayerID) bool {
-	if _, exists := decided[lid]; exists {
-		return false
+func (t *turtle) decodeExceptions(base, ballot *ballotInfo, exceptions types.Votes) {
+	from := base.layer
+	diff := map[types.LayerID]map[types.BlockID]sign{}
+	for vote, bids := range map[sign][]types.BlockID{
+		support: exceptions.Support,
+		against: exceptions.Against,
+	} {
+		for _, bid := range bids {
+			block, exist := t.blockRefs[bid]
+			if !exist {
+				ballot.conditions.votesBeforeBase = true
+				continue
+			}
+			if block.layer.Before(from) {
```

Comment:
compare with base.layer?

Reply:
so the thing here is that i want to find min `from`. and the current approach works because votesBeforeBase is set to true the first time `from` is updated

i can change it to two comparisons

Reply:
changed to two comparisons